### ISSUE TITLE
Log full 3D printing history in Bambu Studio

### ISSUE THREAD

**Issue by wouterfranssens** [bambulab/BambuStudio#2320]
The Bambu Handy app offers the possibility to show the printing history of a single user.
It would be a nice feature that there is also a way to download the full history of what has been printed (by all users).
This way, the used filament could be billed on different projects.
Contrary to the user printer history in the app, users should not be able to delete this full printing history.

**Comment by chrisribe** [comment]
Yes what gives why can I not access this print history in bambu studio ?

**Comment by joshmitcho** [comment]
Any updates on this? @lanewei120

**Comment by SaltWei** [comment]
> It would be a nice feature that there is also a way to download the full history of what has been printed (by all users).
> This way, the used filament could be billed on different projects.

Is there some other user scenario? 
What kind of file format of downloaded history and what information should be necessary in it?
More suggestions would be nice for designing

**Comment by joshmitcho** [comment]
@SaltWei I'd assume a csv would be sufficient, so it would open in Excel
Date, filename, user who printed it, amount and type of each filament used. That would be sufficient for anyone to calculate their own totals for cost recovery/billing.

**Comment by Kayu84** [comment]
A history would be very helpful for various reasons:

- Projects that you do not own, have not printed: average filament used, time and cost calculation
- Projects that you own, have already printed: a nice chart of completed jobs, overview of reprints informations
- overall: also a nice history of the time you have spent printing, how much and which filament you have used...

But please don't add all users (data privacy), I have no idea where it is helpful to know who printed how much filament and how much time spent. Or just make it optional, users who want to share can set this as profile settings.

**Comment by carlstedt1** [comment]
I want this feature so bad! So I can keep my excel print log up to date without much manual work. 

<img width="1523" height="446" alt="Image" src="https://github.com/user-attachments/assets/f3f1315d-adbd-49c0-a69e-7bbc85450358" />
